src/tic_tac_toe_pair/game.clj

```diff
@@ -1,30 +1,53 @@
 (ns tic-tac-toe-pair.game
+  (:require [tic-tac-toe-pair.rules :refer :all]
+            [tic-tac-toe-pair.board :refer :all]
+            [tic-tac-toe-pair.console :refer :all])
   (:gen-class))
 
-(def winning-combinations [[0 1 2] [3 4 5] [6 7 8] 
-                           [0 3 6] [1 4 7] [2 5 8] 
-                           [0 4 8] [2 4 6]])
-                          
-(defn same-tokens? [line]
-  (every? (fn [token] (= token (first line))) line))
-
-(defn build-line [board line-number]
-  (map
-    (fn [square] (get board square))
-    (get winning-combinations line-number)))
-
-(defn get-line-winner [board line-number]
-  (let [line (build-line board line-number)]
-    (if (same-tokens? line)
-      (first line)
-      nil)))
-
-(defn winner-found? [token] (not (nil? token)))
-
-(defn get-winning-token [board]
-  (let [num-combinations (count winning-combinations)]
-    (loop [line-number  0
-           token        nil]
-      (if (or (winner-found? token) (>= line-number num-combinations))
-        token
-        (recur (inc line-number) (get-line-winner board line-number))))))
+(defn initialize-game [] 
+  {:current-token :player-1-token
+   :player-1-token :x
+   :player-2-token :o
+   :board [nil nil nil nil nil nil nil  nil nil]})
+
+(defn update-current-player [game]
+  (assoc game :current-token
+    (if (= (:current-token game) :player-1-token)
+      :player-2-token
+      :player-1-token)))
+
+(defn get-current-token [game] ((:current-token game) game))
+
+(defn update-board [game location]
+  (assoc game :board (fill-location (:board game) location (get-current-token game))))
+
+(defn update-board-and-player [game location]
+  ((comp update-current-player update-board) game location))
+
+(defn update-game [game location]
+  (if (is-move-valid? (:board game) location)
+    (update-board-and-player game location)
+    game))
+
+(defn build-congratulations-message [game token]
+  (str "Congratulations! " (keyword-to-token (token game)) " won the game!"))
```

Comment:
This method seems like it would be a good fit to live in the console namespace, instead.

Reply:
Addressed namespace location for `build-congratulations-message`.